lib/signers/presign.js

```diff
@@ -26,8 +26,9 @@ function signedUrlBuilder(request) {
     }
     request.httpRequest.headers[expiresHeader] = expires;
   } else if (signerClass === AWS.Signers.S3) {
+    var now = request.service ? request.service.getServiceClock() : AWS.util.date.getDate();
```

Comment:
I'm not sure you need to call AWS.util.date.getDate now that skew is handled by service clients. Why not just 'new Date()'?

Reply:
if global `systemClockOffset` is not set, the `AWS.util.date.getDate` will have the same effect as `new Date()`. I think the `AWS.util.date.getDate` can be used as a 'backup' whenever service can not be specified, just like in here (https://github.com/AllanFly120/aws-sdk-js/blob/master/lib/credentials.js#L95)